### PR TITLE
Add shell checks to CI

### PR DESCRIPTION
**Pull request by Praveen2112**
Reference https://github.com/koalaman/shellcheck

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -75,6 +75,14 @@ jobs:
       - name: Web UI Checks
         run: core/trino-main/bin/check_webui.sh
 
+  shell-checks:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v2
+      - name: Shell checks
+        uses: azohra/shell-linter@latest
```
Comment (hashhar) [inline .github/workflows/ci.yml:84]: `latest` causes issues with CI stability IMO.

**Comment by kokosing** [comment]
@Praveen2112 Are you going to fix all the issues?

**Comment by kokosing** [comment]
How to exclude scripts that we do not own?

**Comment by Praveen2112** [comment]
I'm planning to use this action https://github.com/marketplace/actions/shellcheck which provides a way to ignore these actions and also ignore a few checks. WDYT ?

File: core/docker/bin/run-trino [hunk 2]
```diff
@@ -12,4 +12,4 @@ if [[ ${NODE_ID_EXISTS} != 0 ]] ; then
     NODE_ID="-Dnode.id=${HOSTNAME}"
 fi
 
-exec /usr/lib/trino/bin/launcher run --etc-dir /etc/trino ${NODE_ID} "$@"
+exec /usr/lib/trino/bin/launcher run --etc-dir /etc/trino "${NODE_ID}" "$@"
```
Comment (findepi) [inline core/docker/bin/run-trino:15]: that's not correct. the `"` was _intentionally_ missing here

to be even better, you'd need to make the variable an array

**Review by findepi — CHANGES_REQUESTED** [review]

File: .github/bin/retry [hunk 3]
```diff
@@ -10,9 +10,9 @@ x() {
 max_retry_time_seconds=$(( 30 * 60 ))
```
Comment (kokosing) [inline .github/bin/retry:10]: Commits should be order in the following order:
 - fix stuff
 - add CI job
 
 That way all commits have a chance to pass the CI.
 
 Also please extract more commits. I see that you are adding `"` frequently, please extract separate commit with only this so we could see other changes not mixed with it. It is easier to not to overlook things.

File: core/trino-main/bin/check_webui.sh [hunk 4]
```diff
@@ -13,20 +13,20 @@ WEBUI_ROOT="$(pwd)/${BASH_SOURCE%/*}/../src/main/resources/webapp"
...
 # The lockfile is generated code also, and must go through the same process.
 
-pushd $(mktemp -d)
+pushd "$(mktemp -d)"
 
 cp "${WEBUI_ROOT}/src/yarn.lock" .
 cp -r "${WEBUI_ROOT}/dist" .
 
-yarn --cwd ${WEBUI_ROOT}/src/ install
+yarn --cwd "${WEBUI_ROOT}"/src/ install
```
Comment (kokosing) [inline core/trino-main/bin/check_webui.sh:21]: put entire path to `"`. i.e. `"${WEBUI_ROOT}/src/"`, please do the same for others.

File: plugin/trino-hive-hadoop2/bin/common.sh [hunk 5]
```diff
@@ -56,15 +56,18 @@ function stop_unnecessary_hadoop_services() {
 function expand_compose_args() {
     local files=( "${@}" )
     local compose_args=""
-    for file in ${files[@]}; do
+    for file in "${files[@]}"; do
```
Comment (kokosing) [inline plugin/trino-hive-hadoop2/bin/common.sh:59]: does it work? Will get a list of values or a single value?

**Comment by bitsondatadev** [comment]
:wave: @Praveen2112 - this PR is inactive and doesn't seem to be under development. If you'd like to continue work on this at any point in the future, feel free to re-open.

**Pull request closed without merging**